src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +111,29 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        clearInterval(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private setPollingMechanism(gatewayId: string | undefined, submitForm: any) {
+        this._pollingInterval = setInterval(async () =>  {
+            try {
+                if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+                    const res = await this._paypalCommerceRequestSender.getOrderStatus();
+                    if (res.status.toLowerCase() === APPROVED) {
+                        clearInterval(this._pollingInterval);
+                        this._tokenizePayment({orderID: this._paypalCommercePaymentProcessor.getOrderId()}, submitForm);
+                    }
+                }
+            } catch (e) {
+                clearInterval(this._pollingInterval);
+            }
+        }, 3000);
```

Comment:
maybe this can be a const at the top of the file, like `APPROVED`

Reply:
fixed